File: AlternativeQL/src/org/ql/ast/literal/BooleanLiteral.java
package org.ql.ast.literal;

import org.ql.ast.Node;

public class BooleanLiteral implements Node {
    private boolean booleanLiteral;

    public BooleanLiteral(boolean booleanLiteral) {
        this.booleanLiteral = booleanLiteral;
    }

    public boolean getBooleanLiteral() {
        return booleanLiteral;
    }
}

File: OffByOne/src/OffByOne.Ql/Ast/Literals/BooleanLiteral.cs
namespace OffByOne.Ql.Ast.Literals
{
    using System;

    using OffByOne.Ql.Ast.Literals.Base;
    using OffByOne.Ql.Values;
    using OffByOne.Ql.Visitors.Contracts;

    public class BooleanLiteral : Literal
    {
        public BooleanLiteral(bool value)
        {
            this.Value = new BooleanValue(value);
        }

        public BooleanLiteral(string value)
        {
            if (value == null)
            {
                throw new ArgumentNullException(nameof(value), "Literal must have a value.");
            }

            this.Value = new BooleanValue(value);
        }

        public BooleanValue Value { get; private set; }

        public override TResult Accept<TResult, TContext>(
            IExpressionVisitor<TResult, TContext> visitor,
            TContext context)
        {
            return visitor.Visit(this, context);
        }
    }
}
